connector/src/main/java/org/jboss/as/connector/subsystems/datasources/Util.java

```diff
@@ -46,7 +46,7 @@ public class Util {
      */
     public static String getJndiName(final OperationContext context, final ModelNode modelNode) throws OperationFailedException {
         final String rawJndiName = JNDI_NAME.resolveModelAttribute(context, modelNode).asString();
-        return cleanJndiName(rawJndiName, modelNode.hasDefined(USE_JAVA_CONTEXT.getName()) && modelNode.get(USE_JAVA_CONTEXT.getName()).asBoolean());
+        return cleanJndiName(rawJndiName, modelNode.hasDefined(USE_JAVA_CONTEXT.getName()) && USE_JAVA_CONTEXT.resolveModelAttribute(context, modelNode).asBoolean());
```

Comment:
The modelNode.hasDefined(USE_JAVA_CONTEXT.getName()) is wrong too. USE_JAVA_CONTEXT has a default value of 'true', so that call is disabling the use of that default value.